overlord/hookstate/ctlcmd/is_connected.go

```diff
@@ -87,6 +111,34 @@ func (c *isConnectedCommand) Execute(args []string) error {
 		return fmt.Errorf("internal error: cannot get connections: %s", err)
 	}
 
+	var otherSnap *snap.Info
+	if c.AppArmorLabel != "" {
+		if !isConnectedPidCheckAllowed(info, plugOrSlot) {
+			return fmt.Errorf("cannot use --apparmor-label check with %s:%s", snapName, plugOrSlot)
+		}
+		name, _, _, err := apparmor.DecodeLabel(c.AppArmorLabel)
+		if err != nil {
+			return &UnsuccessfulError{ExitCode: 10}
```

Comment:
independently of the values we probably want to introduce constants:  

notASnapCode
classicSnapCode